### PR TITLE
[Synthetics] Support SSL

### PR DESCRIPTION
**Pull request by btoueg**
- Add support for ssl tests
 - Add acceptance test
 - Update doc
 - Have subtype defaults to http for api tests
 - Make sure existing TF files won't diff

### CODE REVIEW

**Comment by nmuesch** [comment]
Hey @btoueg  thanks for the PR! The golang client was released + bumped in this provider. Can you rebase from master and we can work on getting this reviewed?

File: datadog/resource_datadog_synthetics_test_.go [hunk 1]
```diff
@@ -316,6 +359,9 @@ func newSyntheticsTestFromLocalState(d *schema.ResourceData) *datadog.Synthetics
 		minLocationFailed, _ := strconv.Atoi(attr.(string))
 		options.SetMinLocationFailed(minLocationFailed)
 	}
+	if attr, ok := d.GetOk("options.accept_self_signed"); ok {
+		options.SetAcceptSelfSigned(attr.(string) == "true")
```
Comment (nmuesch) [inline datadog/resource_datadog_synthetics_test_.go]: Any reason not to just assign this to `attr.(string)` Should we also safely type assert this?

File: website/docs/r/synthetics.html.markdown [hunk 2]
```diff
@@ -43,6 +45,38 @@ resource "datadog_synthetics_test" "foo" {
 }
 ```
 
+## Example Usage (Synthetics SSL test)
+
+Create a new Datadog Synthetics API/SSL test on example.org
+
+```hcl
+resource "datadog_synthetics_test" "foo" {
```
Comment (nmuesch) [inline website/docs/r/synthetics.html.markdown]: Lets name this something else to not clash with the other example named `foo`

File: datadog/resource_datadog_synthetics_test_.go (hunk 1 above)
Comment (btoueg) [reply datadog/resource_datadog_synthetics_test_.go]: Not sure, can't wrap my head around it. It's value is "1" or "0" in TF 0.11, even though is never set to such value

**Comment by btoueg** [comment]
> two successive plans cause an un-empty diff

Indeed, that's the case in TF 0.11. I've handled it, let me know

**Comment by nmuesch** [comment]
Thanks for the iterations here! Looks good to me, I'll go ahead and merge. Thanks for this 🎉🚀

**Pull request merged**